### PR TITLE
Reorganize battle/; rename most "header" misnomers; gfx/pics → gfx/pokemon

### PR DESCRIPTION
**Pull request by roukaour**
I left `map_header` and `map_header_2` alone because (1) `map_data` and `map_data_2` sound generic, (2) pokered also calls them headers, and (3) technically `*_MapScriptHeader` and `*_MapEventHeader` aren't headers either. (`MapEventHeader` is more like `MapScriptFooter`.) Plus the fact that `MapEventHeader` (and `musicheader`s) come right before their data is convention/coincidence; they only need to be in the same bank. audio/cries.asm's `musicheader`s are all grouped together, separate from all the cry channel data. Anyway, updating the map-related labels can be done along with refactoring the `maps/` directory structure, however we decide to do that.

### CODE REVIEW

File: wram.asm [hunk 1]
```diff
@@ -2941,21 +2941,16 @@ wBattleAnimTemp0:: db
...
-wBattleAnimTemp4:: db
-wBattleAnimTemp5:: db
-wBattleAnimTemp6:: db
-wBattleAnimTemp7:: db
-wBattleAnimTemp8:: db
 
 NEXTU ; d419
 wBattleAnimTempOAMFlags:: db
-	ds 1
+wBattleAnimTempField02:: db
```
Comment (yenatch) [inline wram.asm]: 02?

File: engine/mail.asm [hunk 2]
```diff
@@ -397,11 +397,11 @@ MailboxPC: ; 0x44806
 
 	ld a, [wCurMessageIndex]
 	ld [wMenuCursorBuffer], a
-	ld a, [OBPals + 8 * 6]
+	ld a, [OBPals palette 6]
 	ld [wMenuScrollPosition], a
 	call ScrollingMenu
 	ld a, [wMenuScrollPosition]
-	ld [OBPals + 8 * 6], a
+	ld [OBPals palette 6], a
```
Comment (yenatch) [inline engine/mail.asm]: these seem wrong

File: maps/AzaleaGym.asm [hunk 3]
```diff
@@ -91,35 +91,35 @@ TrainerTwinsAmyandmay2:
 	closetext
 	end
 
-TrainerBug_catcherbenny:
-	trainer EVENT_BEAT_BUG_CATCHER_BENNY, BUG_CATCHER, BUG_CATCHER_BENNY, Bug_catcherbennySeenText, Bug_catcherbennyBeatenText, 0, .AfterScript
+TrainerBugCatcherbenny:
+	trainer EVENT_BEAT_BUG_CATCHER_BENNY, BUG_CATCHER, BUG_CATCHER_BENNY, BugCatcherbennySeenText, BugCatcherbennyBeatenText, 0, .AfterScript
```
Comment (yenatch) [inline maps/AzaleaGym.asm]: benny

File: engine/mail.asm (hunk 2 above)
Comment (roukaour) [reply engine/mail.asm]: Can you please elaborate? `make compare` is OK, and that's been a common pattern for loading Nth palettes.

Comment (yenatch) [reply engine/mail.asm]: Why would the menu scroll position be the first byte of the palette?

File: wram.asm (hunk 1 above)
Comment (roukaour) [reply wram.asm]: It corresponds to `BATTLEANIMSTRUCT_02`.

**Pull request merged**